File: src/main/java/com/adven/concordion/extensions/exam/files/FilesResultRenderer.java
package com.adven.concordion.extensions.exam.files;

import org.concordion.api.Element;
import org.concordion.api.listener.*;

public class FilesResultRenderer implements AssertEqualsListener, AssertTrueListener, AssertFalseListener {

    public void failureReported(AssertFailureEvent event) {
        Element element = event.getElement();
        element.addStyleClass("rest-failure");

        Element expected = new Element("del");
        expected.addStyleClass("expected");
        element.moveChildrenTo(expected);
        element.appendChild(expected);
        expected.appendNonBreakingSpaceIfBlank();

        Element actual = new Element("ins");
        actual.addStyleClass("actual");
        actual.appendText(convertToString(event.getActual()) +
                (event.getExpected() == null ? " (surplus)" : ""));
        actual.appendNonBreakingSpaceIfBlank();

        element.appendText("\n");
        element.appendChild(actual);
    }

    public void successReported(AssertSuccessEvent event) {
        event.getElement().addStyleClass("rest-success").appendNonBreakingSpaceIfBlank();
    }

    private String convertToString(Object object) {
        return object == null ? "(absent)" : "" + object;
    }
}

File: exam-files/src/main/java/com/adven/concordion/extensions/exam/files/FilesResultRenderer.kt
package com.adven.concordion.extensions.exam.files

import com.adven.concordion.extensions.exam.core.ExamResultRenderer
import com.adven.concordion.extensions.exam.core.toStringOr
import org.concordion.api.listener.AssertFailureEvent

class FilesResultRenderer : ExamResultRenderer() {

    override fun actualText(event: AssertFailureEvent): String {
        return event.actual.toStringOr("(absent)") + if (event.expected == null) " (surplus)" else ""
    }
}
